types/datum.go

```diff
@@ -1468,6 +1469,14 @@ func (d *Datum) convertToMysqlDecimal(sc *stmtctx.StatementContext, target *Fiel
 // ProduceDecWithSpecifiedTp produces a new decimal according to `flen` and `decimal`.
 func ProduceDecWithSpecifiedTp(dec *MyDecimal, tp *FieldType, sc *stmtctx.StatementContext) (_ *MyDecimal, err error) {
 	flen, decimal := tp.GetFlen(), tp.GetDecimal()
+
+	//If string to decimal, determine the precision for each string
```

Comment:
Is there any possible that `(tp.GetFlag() & mysql.StringToDecimalFlag)` is true and `flen != UnspecifiedLength && decimal != UnspecifiedLength ` is false?

Reply:
NO,  None of these types （mysql.TypeString, mysql.TypeVarchar, mysql.TypeTinyBlob, mysql.TypeMediumBlob, mysql.TypeLongBlob, mysql.TypeBlob） GetFlen（） == -1

Reply:
you'd better to decrease cyclomatic complexity. I mean that you added `if {}` block before origin `if flen != UnspecifiedLength && decimal != UnspecifiedLength {}`, the code looks a little hard to read and maintain.

Reply:
down

Reply:
done